### PR TITLE
Create maf to vcf

### PR DESCRIPTION
**Pull request by zrm22**
## Description

Initial code for creating a g and hvcf file from the MAF file

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Added in functionality to create h.vcf file from a MAF file.
```

### CODE REVIEW

**Review by pjbradbury — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+import net.maizegenetics.phgv2.utils.Position
+import net.maizegenetics.phgv2.utils.createHVCFRecord
+import net.maizegenetics.phgv2.utils.exportVariantContext
+import net.maizegenetics.phgv2.utils.getChecksumForString
+import java.io.File
+
+class CreateMafVcf : CliktCommand(help = "Create gVCF and hVCF from Anchorwave MAF files") {
+
+    val bed by option(help = "BED file")
+        .default("")
```
Comment (tcasstevens) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt:21]: Still seems like putting required() would be sufficant?

Comment (zrm22) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt:21]: I agree, but I have tried this and it does not work unless the default is there.

File: src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+            }
+        }
+
+    /**
+     * Function to create the ASM hVCF and gVCF.
+     * It will first use Biokotlin to build the gVCF and then will use the BED file to extract out the hVCF information.
+     */
+    fun createASMHvcfs(dbPath: String, bedFileName: String, referenceFileName: String, mafDirName: String, outputDirName: String) {
+        //load the bed file into some data structure
+//        val ranges = bedfileToSRangeSet(bedFileName,referenceFileName)
```
[the remark this answers is not in the document]
Comment (zrm22) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt:66]: I am going to leave it for now as I would like to use a Biokotlin class, but SRanges are too heavy.  Once I update the MAFToGVCF code in biokotlin I can add this in.

File: src/test/kotlin/net/maizegenetics/phgv2/cli/CreateMafVCFTest.kt [hunk 3]
```diff
@@ -0,0 +1,477 @@
...
+import net.maizegenetics.phgv2.utils.createSNPVC
+import net.maizegenetics.phgv2.utils.getChecksumForString
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import java.io.File
+import kotlin.test.*
+
+@ExtendWith(TestExtension::class)
+class CreateMafVCFTest {
+    companion object {
```
[the remark this answers is not in the document]
Comment (zrm22) [reply src/test/kotlin/net/maizegenetics/phgv2/cli/CreateMafVCFTest.kt:19]: Not yet.  When I transfer over the AGC code, I will need to use this to build the AGC record before the unit tests start.

**Review by tcasstevens — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/cli/SetupEnvironment.kt [hunk 4]
```diff
@@ -16,7 +16,7 @@ import java.util.stream.Collectors
  * Check if this environment already exists.  If an error occurs when
  * creating it, do we assume it exists?
  */
-class SetupEnvironment : CliktCommand() {
+class SetupEnvironment : CliktCommand(help="create conda environment for PHG") {
```
Comment (btmonier) [inline src/main/kotlin/net/maizegenetics/phgv2/cli/SetupEnvironment.kt:19]: Maybe this was discussed in another PR, but will this also exported into the `Phg().subcommands()` method?

**Review by btmonier — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/cli/SetupEnvironment.kt (hunk 4 above)
Comment (zrm22) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/SetupEnvironment.kt:19]: Yes looks like this was forgotten about.  I will add it in.

**Pull request merged**